classes/xorggroupexsession.php

```diff
@@ -49,6 +49,24 @@ public function startAvailableAuth()
             S::set('loginX', $url);
         }
 
+        if (S::user()->type == 'xnet') {
+            // If this is a Xnet account and it is still 'pending', it is now active.
+            $pending = XDB::fetchOneCell('SELECT COUNT(*) FROM accounts
+                                           WHERE uid = {?} AND state = \'pending\' AND type = \'xnet\'',
+                                         S::user()->id());
+            if $pending > 0 {
```

Comment:
PHP needs parentheses ;)
Moreover the ``User`` class provides ``$user->isActive()``, which is implemented as ``return $this->state == 'active';``. We can't use this function (because an account can also be disabled, so "not active" does not mean "pending"), but using ``S::user()->state == 'pending'`` should work... Are there any issues with this?